lib/resty/core/regex.lua

```diff
@@ -79,24 +103,36 @@ local ngx_lua_ffi_init_script_engine
 local ngx_lua_ffi_compile_replace_template
 local ngx_lua_ffi_script_eval_len
 local ngx_lua_ffi_script_eval_data
--- PCRE 8.43 on macOS introduced the MAP_JIT option when creating memory region
--- used to store JIT compiled code, which does not survive across `fork()`,
--- causing further usage of PCRE JIT compiler to segfault in worker processes
+
+-- PCRE 8.43 on macOS introduced the MAP_JIT option when creating the memory
+-- region used to store JIT compiled code, which does not survive across
+-- `fork()`, causing further usage of PCRE JIT compiler to segfault in worker
+-- processes.
 --
--- this flag prevents any regex used in init phase to be JIT compiled or cached
--- when running under macOS even if the user requests so. caching is disabled
--- to prevent further calls of same regex in worker to have poor performance.
-local pcre_map_jit_fix = true
+-- This flag prevents any regex used in the init phase to be JIT compiled or
+-- cached when running under macOS, even if the user requests so. Caching is
+-- thus disabled to prevent further calls of same regex in worker to have poor
+-- performance.
+--
+-- TODO: improve this workaround when PCRE allows for unspecifying the MAP_JIT
+-- option.
+local no_jit_in_init
 
+if jit.os == "OSX" then
+    pcre_ver = ffi_string(pcre_ver)
```

Comment:
It's not a great idea to use the same Lua variable name for values of completely differently types.

Reply:
I have to say I concur.